tests/test_file_operations.py

```diff
@@ -583,7 +587,91 @@ def test_configure_not_absolute():
     # Encode the pickled dictionary with Base64
     encoded_dict = base64.b64encode(pickled_dict)
     parser = get_parser()
-    cmd = f"configure {tagged_file} {tagged_file} {tag} {encoded_dict}"
+    cmd = f"configure_file {tagged_file} {tagged_file} {tag} {encoded_dict}"
+    args = cmd.split()
+
+    with pytest.raises(SystemExit) as e:
+        parser.parse_args(args)
+
+    assert isinstance(e.value.__context__, argparse.ArgumentError)
+    assert "invalid _abspath value" in e.value.__context__.message
+
+
+@pytest.mark.parametrize(
+    ["param_dict", "error_type"],
+    [
+        pytest.param(
+            {"PARAM0": "param_value_1", "PARAM1": "param_value_2"},
+            "None",
+            id="correct dict",
+        ),
+        pytest.param(
+            ["list", "of", "values"],
+            "TypeError",
+            id="incorrect dict",
+        ),
+        pytest.param({}, "ValueError", id="empty dict"),
+    ],
+)
+def test_configure_directory(test_dir, fileutils, param_dict, error_type):
+    """Test configure_directory operation with correct parameter dictionary, empty dicitonary, and an incorrect type"""
+    tag = ";"
+    config = get_gen_file(fileutils, "tag_dir_template")
+    path = shutil.copytree(config, test_dir, dirs_exist_ok=True)
+    assert osp.isdir(path)
+
+    # Pickle the dictionary
+    pickled_dict = pickle.dumps(param_dict)
+    # Encode the pickled dictionary with Base64
+    encoded_dict = base64.b64encode(pickled_dict).decode("ascii")
+
+    parser = get_parser()
+    cmd = f"configure_directory {path} {path} {tag} {encoded_dict}"
+    args = cmd.split()
+    ns = parser.parse_args(args)
+
+    if error_type == "ValueError":
+        with pytest.raises(ValueError) as ex:
+            file_operations.configure_directory(ns)
+            assert "param dictionary is empty" in ex.value.args[0]
+    elif error_type == "TypeError":
+        with pytest.raises(TypeError) as ex:
+            file_operations.configure_directory(ns)
+            assert "param dict is not a valid dictionary" in ex.value.args[0]
+    else:
+        file_operations.configure_directory(ns)
+    if error_type == "None":
```

Comment:
Unnecessary check? Seems like this has to be the case to enter the `else` block here